File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/holder/pipeline/XMLBase.java
package com.kite9.k9server.adl.holder.pipeline;

import java.net.URI;

import org.springframework.http.HttpHeaders;

public interface XMLBase {

	URI getUri();

	/**
	 * Returns the HTTPHeaders that were responsible for loading this 
	 * ADL.  Useful for passing around credentials.
	 */
	HttpHeaders getRequestHeaders();

	/**
	 * Returns the XML.
	 */
	String getXMLString();

}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/pipeline/adl/holder/pipeline/XMLBase.kt
package com.kite9.pipeline.adl.holder.pipeline

import com.kite9.pipeline.uri.K9URI

/**
 * Base interface that captures where an xml document came from, and it's content.
 */
interface XMLBase {

    val uri: K9URI?

    /**
     * Returns the HTTPHeaders that were responsible for loading this
     * ADL.  Useful for passing around credentials.
     */
    val requestHeaders: Map<String, List<String>>

    /**
     * Returns the XML.
     */
    val asString: String?
}